fractal_server/app/runner/__init__.py

```diff
@@ -110,19 +115,36 @@ async def submit_workflow(
     logger.info(f"output_path: {output_path}")
     logger.info(f"input metadata: {input_dataset.meta}")
     logger.info(f"START workflow {workflow.name}")
-    output_dataset.meta = await process_workflow(
-        workflow=workflow,
-        input_paths=input_paths,
-        output_path=output_path,
-        input_metadata=input_dataset.meta,
-        username=username,
-        workflow_dir=WORKFLOW_DIR,
-        logger_name=logger_name,
-        worker_init=worker_init,
-    )
+    job.working_dir = WORKFLOW_DIR.as_posix()
+    job.status = JobStatusType.RUNNING
+    db_sync.merge(job)
+    db_sync.commit()
+    try:
+        output_dataset.meta = await process_workflow(
+            workflow=workflow,
+            input_paths=input_paths,
+            output_path=output_path,
+            input_metadata=input_dataset.meta,
+            username=username,
+            workflow_dir=WORKFLOW_DIR,
+            logger_name=logger_name,
+            worker_init=worker_init,
+        )
+
+        logger.info(f'END workflow "{workflow.name}"')
+        close_job_logger(logger)
```

Comment:
Logger should be closed also in case of failed execution.
In that case, we can also replicate the `job.log` contents in the logger.